### ISSUE TITLE
SkyCoord.spherical_offsets_to crashes for vector data

### ISSUE THREAD

**Issue by fardal** [astropy/astropy#5619]
```
import numpy as np
from astropy.coordinates import SkyCoord
c0s = SkyCoord.from_name('M31')
c1s = SkyCoord.from_name('M33')
c0 = SkyCoord(ra=np.ones(100)*c0s.ra, dec=np.ones(100)*c0s.dec, frame='icrs')
c1 = SkyCoord(ra=np.ones(100)*c1s.ra, dec=np.ones(100)*c1s.dec, frame='icrs')
print c0.spherical_offsets_to(c1)
```

This returns a fairly perplexing error:
```
AttributeError: 'Quantity' object has no 'hypot' member
```
The same operation with `c0s` and `c1s` (which have scalar content) works fine.

This is with astropy 1.2.1, numpy 1.11, python 2.7.12

**Comment by fardal** [comment]
I'm guessing the reason it gives such a weird error is distinct from the fundamental problem.

The example above gets to
```
astropy/coordinates/builtin_frames/skyoffset.py(130)reference_to_skyoffset()
-> mat2 = rotation_matrix(-origin.lat, 'y')
```

My guess at the fundamental problem is that `origin.lat` is a vector-valued `Quantity`.
I think `rotation_matrix` is expecting scalar input - internally it uses `np.matrix` which only makes 2-d matrices.

Then, if `rotation_matrix` did give a more than 2-d matrix out, I wonder - would the rest of the method be able to handle it?  

In fact `rotation_matrix` does manage to give a very strange mutant matrix back in this case without raising an exception, I suspect entirely by accident.  This creates the weird error later on.

**Comment by adrn** [comment]
Looks like this is the same as #5163 -- I think this is fixed in v1.3. Would you mind trying with the v1.3 release candidate? (you should be able to do: `pip install astropy==1.3rc1`, but I would do this in a clean conda or virtual environment)

**Comment by fardal** [comment]
Seems to work!  I totally missed #5163.
